### PR TITLE
Fonts alignment

### PR DESCRIPTION
**Pull request by MichelleXBai**
https://jira.zopa/browse/OBAU-26623

### CODE REVIEW

File: src/components/templates/ProductTemplate/ProductTemplateTitle/ProductTemplateTitle.tsx [hunk 1]
```diff
@@ -51,17 +51,22 @@ export function ProductTemplateTitle({
     <ProductTemplateTitleBackground data-automation={dataAutomation} theme={theme}>
       <ProductTemplateTitleContainer>
         <ProductTemplateTitleInnerContainer className="pt-7 m:pt-9 mx-6 m:mx-0">
-          {title && (
```
Comment (admmasters) [inline src/components/templates/ProductTemplate/ProductTemplateTitle/ProductTemplateTitle.tsx:54]: Always avoid `&&` in jsx

**Review by OlenaKashuba — COMMENTED** [review]
changeset needed!

File: src/components/molecules/DropdownFiltered/Option.ts [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 }
+interface OptionThemeProps extends AppThemeProps, OptionProps {}
 
-const Option = styled.div<OptionProps & { theme: AppTheme }>`
+const Option = styled.div<OptionThemeProps>`
   cursor: pointer;
   padding: 8px;
-  line-height: 32px;
-  font-family: ${({ theme }) => theme.typography.primary};
+  line-height: ${({ theme }: OptionThemeProps) => theme.typography.lineHeight.body ?? '32px'};
```
[the remark this answers is not in the document]
Comment (OlenaKashuba) [reply src/components/molecules/DropdownFiltered/Option.ts:14]: i vote for required. If it breaks early, its easier to spot than it falls back to a wrong value and we don't notice

File: src/components/styles/Theme/ThemeProvider.tsx [hunk 3]
```diff
@@ -281,14 +297,10 @@ export interface TypographyTheme {
   };
   weights: {
     regular: number;
+    medium?: number;
```
Comment (OlenaKashuba) [inline src/components/styles/Theme/ThemeProvider.tsx:300]: is it safer to have those required?

Comment (MichelleXBai) [reply src/components/styles/Theme/ThemeProvider.tsx:300]: We don't have medium in zopa theme, but  happy to add it and set it to 500 if it makes more sense?

File: src/components/molecules/DropdownFiltered/Option.ts [hunk 4]
```diff
@@ -1,24 +1,25 @@
...
+  font-family: ${({ theme }: OptionThemeProps) => theme.typography.primary};
   font-size: ${typography.sizes.text.body};
-  font-weight: 600;
+  font-weight: ${({ theme }: OptionThemeProps) => theme.typography.weights.regular ?? '600'};
```
[the remark this answers is not in the document]
Comment (MichelleXBai) [reply src/components/molecules/DropdownFiltered/Option.ts:17]: Updated

File: src/components/molecules/DropdownFiltered/Option.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichelleXBai) [reply src/components/molecules/DropdownFiltered/Option.ts:14]: Updated

File: src/components/styles/Theme/ThemeProvider.tsx (hunk 3 above)
Comment (MichelleXBai) [reply src/components/styles/Theme/ThemeProvider.tsx:300]: Updated. Also updated bold to requried as well.

**Review by admmasters — APPROVED** [review]
Great job!

**Review by zopa-charlotte-walker — APPROVED** [review]

**Pull request merged**